### PR TITLE
Update game_menu_impl.h

### PR DESCRIPTION
**Pull request by manichandra3**
Improve menu initialization and error handling

- Refactored font loading to a separate function
- Handled font loading failures with exception
- Moved duplicate item processing outside the try block
- Updated variable naming for consistency
- Cleaned up unnecessary console outputs

### CODE REVIEW

**Comment by ParadoxZero** [comment]
~You forgot to add the cpp file and its changes.~
Nvm, saw the inline methods.

File: src/game_menu_impl.h [hunk 1]
```diff
@@ -12,42 +12,60 @@
...
+
+    private:
+        void setMenu();
+        void drawMenu();
+        void writeText(std::string str, sf::Font* font, unsigned int size, float x,
+                       float y, const Color color);
+
+        void loadFont() {
+            // Implement font loading logic, throw std::runtime_error on failure
+            if (!_style.ItemFont->loadFromFile("path/to/your/font.ttf")) {
```
Comment (ParadoxZero) [inline src/game_menu_impl.h:49]: This is fundamentally wrong. Game menu class should receive the fonts already loaded by the host. 

Even if we need to add the API to load it from the `GameMenu` class, we can't hardcode a path like this.

File: src/game_menu_impl.h [hunk 2]
```diff
@@ -12,42 +12,60 @@
...
+                throw std::runtime_error("Failed to load font");
```
Comment (ParadoxZero) [inline src/game_menu_impl.h:50]: I don't like `Exceptions` period. They are unpredictable and hard to debug. Throwing exception explicitly in constructor is bad design in my book.

While it's not relevant (as I don't think we should be loading font in the class), if at all we have failable initialization. We should create some static factory method to create the object instead.

File: src/game_menu_impl.h [hunk 3]
```diff
@@ -12,42 +12,60 @@
...
+            }
+        }
+
+        void processItems(const std::vector<MenuItem>& items) {
+            for (auto& menu_item : items) {
+                _items.push_back({ menu_item, {0, 0} });
+                std::cout << " Processing item " << menu_item.name << std::endl;
```
Comment (ParadoxZero) [inline src/game_menu_impl.h:57]: Remove these `couts` if possible. They are remains from a debugging session.

**Comment by ParadoxZero** [comment]
It's not clear from the PR, which duplication did you remove? The process items logic only existed in the constructor from what I can see.

Also check if you have any unpublished commits, since the `couts` are still there.

**Comment by ParadoxZero** [comment]
Can you explain why you think we should load the fonts in GameMenu and not in the host?

**Review by ParadoxZero — CHANGES_REQUESTED** [review]

**Comment by manichandra3** [comment]
Sorry for the lazy PR, I am still new to development in C++ and I thought loading the fonts in the GameMenu will help with separation of concerns but I just realised that it is the complete opposite. Thank you for taking the time to guide me.

**Comment by ParadoxZero** [comment]
@manichandra3 no worries, we all are always learning. Feel free to fix this PR and reopening it. Reviewing it, I realized some potential issues in the lib. I have added them to list. You can take a stab at them too if you want, I review them for you :)

**Pull request closed without merging**